File: GoTravelServidor/src/main/java/com/gotravel/server/model/Contratacion.java
package com.gotravel.server.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "contratacion")
@NoArgsConstructor
@AllArgsConstructor
public class Contratacion {

    public Contratacion(String id, Pago pago, String fecha) {
        this.id = id;
        this.pago = pago;
        this.fecha = fecha;
    }

    @Id
    @Column(name = "id_contratacion", nullable = false)
    private String id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_servicio", nullable = false)
    private Servicio servicio;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

    @Column(name = "fecha", nullable = false, columnDefinition = "DATE")
    private String fecha;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_etapa")
    private Etapa etapa;

    @OneToOne(fetch = FetchType.LAZY, optional = false, cascade = CascadeType.ALL)
    @JoinColumn(name = "id_pago", nullable = false)
    private Pago pago;

}

File: GoTravelMobile/app/src/main/java/com/gotravel/mobile/data/model/Contratacion.kt
package com.gotravel.mobile.data.model

import kotlinx.serialization.Serializable
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Serializable
data class Contratacion(
    val id: Int,
    val servicio: Servicio,
    val usuario: Usuario,
    val fecha: String,
    val etapa: Etapa,
    val pago: Pago
)